common/djangoapps/util/date_utils.py

```diff
@@ -10,9 +10,46 @@
 import six
 from django.utils.translation import pgettext, ugettext
 from pytz import UnknownTimeZoneError, timezone, utc
+from xmodule.course_module import DEFAULT_START_DATE
+from openedx.core.djangoapps.schedules.models import Schedule
 
 
-def get_default_time_display(dtime):
+def is_unset(date_field_value):
+    if isinstance(date_field_value, timedelta):
```

Comment:
just curious, did you consider using a custom class to support both date types and encapsulate the `isinstance` logic?

Reply:
I didn't, really. Partly, that was just the nature of doing it as a spike; I was just throwing stuff at the wall until things stuck. Even now, though, I'm not sure that the `ininstance` logic is repeated consistently enough to make a custom class the right choice. I'm certainly open to the discussion on that, though.